### DISCUSSION TITLE
Which BMS for Parallel Battery Packs

### DISCUSSION THREAD

**Discussion by SlapJackNpNp** [Louisvdw/dbus-serialbattery#639] [General]
Hey everyone, i just finished my first Battery Pack with a Daly BMS and lurk arround here to watch the Discussions since i'm no coder unfortunately. 

Reading in the Discussions regarding Dally it seems like it might not be the best BMS to use for a Parallel 48V LifePo4 Battery pack System. So i wanted to ask what of the available BMS Systems to this dirver would be more handy for such a project. 

Thanks for the suggestions.

**Comment by transistorgit** [comment]
I use 2 daly in parallel  and don't see a problem (although I'm a coder and had to rework parts of the driver to make it more stable and useful)

**Reply by SlapJackNpNp** [reply]
I was planning on using 2 dalys, but the batterie aggregator from doc. Gigavolt does not work for daly as far as i'm aware. I saw that you contributed alot to the daly driver and helped making it possible to distinguish two dalys. Unfortunately i'm no software developer, so rewriting the code is not an option for me. 

If i'm not mistaken tho, Dr. Gigavolt said that the problem with dalys not working with his driver is not a problem of his driver but rather of the daly driver in the serial battery driver? 

Did you by any chance test if 2 dalys with the new pre release drivers are working with dr gigavolt? Or did you use another batterie aggregator? 

Thank you very much

**Reply by transistorgit** [reply]
Hi,
I use 2 dalys since months now without problems. There where some hickups during implementing new features, but this is normal for beta versions. There will be a new "official" release of Louisvdw/dbus-serialbattery end of may, you can use that as soon as it is out.

brgds

**Reply by SlapJackNpNp** [reply]
Oh thats great news. Thank you very much for the Update, looking forward to try this!

**Comment by dfch** [comment]
I am curently using JK-BMS with 2 16s batteries via RS485 and the [BatteryAggregator](https://github.com/pulquero/BatteryAggregator) to combine both (as of now without problems).

**Reply by SlapJackNpNp** [reply]
Thanks for the feedback. I found other people doing it with jk-bms as well. It seems to be one of the more used BMS systems that work with most drivers. Gonna have a look at them

**Comment by CaptKrisp** [comment]
Recently I built and added parallel packs to both my 48V and 12V systems.
So, now I have 2 48V packs in parallel with Daly BMSs and 2 12V packs in parallel with Daly BMSs.  (My BMSs are not the ones designed for "parallel" connections but they work without problems ... so long as the parallel packs are not more than a few tenths of a volt difference at any time.)
SerialBattery works well with 2 Daly BMSs connected directly to rPi but if connected via USB hub then Python doesn't like it.
BatteryAggregator also works well in both systems.

**Reply by SlapJackNpNp** [reply]
Thank you very much for the reply. Just to make sure i get it right. You use the battery aggregator by dr. Gigavolt with the newest. 1.xx serial battery driver? 

Does the rpi have more usb connections then the cerbo? Or how was your solution foe the lack of usb connectors on the rpi if hubs are not working? 

Thank you very much!

**Reply by transistorgit** [reply]
I connected them by a 5m active USB extension cable and an usb hub with galvanic isolation, that works without issues. I once burned a pi (had to resolder usb driver chip) by connecting the BMS directly. I think there may be high currents on the ground wire if the battery deliveres high current? My battery is not grounded, just connected to +/- at Multiplus.

Regarding having no parallel module: I wonder what will happen when 1 of the 2 BMS switches off (maybe because of a bad temperature sensor, I have -40°C readouts occasionally). Then the other one can be used to, say, deplete one battery. Then you have one full and  one empty battery. What happens when the first BMS closes contact again? Will it burn, explode, or is it fast enough to detect overcurrent and switch off?

**Reply by SlapJackNpNp** [reply]
I don't think there is a 100% safe and automatic solution. Maybe the new daly modules but i haven't found many information regarding what they do. 

Currently i monitor the batteries regularily and also limit the current of each pack to a, what i think, safe current. If the currwnt exceeds the daly turns off. Additionally i have a fuse in each pack to have a fall back level.

Once indetect that there is to much difference in the packs i will reload them individually to full bevore i connect them parallel again.

**Reply by CaptKrisp** [reply]
I am using pulquero/BatteryAggregator as installed from the PackageManager (installed via SetupHelper) in the Venos OS settings page.
The rPi has 4 USB ports.  I don't have a Cerbo so not familiar with the number of ports on that unit.
My rPi USB connections are:
1) Victron inverter via MK3 interface
2) Daly BMS
3) Daly BMS
4) USB hub
4a) Victron solar charge controller 1
4b) Victron solar charge controller 2
4c) USB thumb drive

**Reply by CaptKrisp** [reply]
@transistorgit yes, when 1 BMS in parallel packs disables either charge or discharge there is concern or high current when the BMS enables those functions.  I have yet to develop a method to deal with it.  I am planning to use Node-red to resolve it but haven't had time to develop a plan of action/implementation.
 I have had a couple occasions when 1 of the cells was high and out of balance in 1 of the 12V packs so the BMS shutdown the charge and discharge of the pack.  This was when there was no active balancer installed.  A couple weeks ago an active balancer was installed and once the cells were balanced (took 2 days) I connected the pack to the DC bus and have not yet had any issues.
In my system at this time the only reasons a BMS should disable the charge/discharge is when a cell voltage is too high/low or when the temp is too high/low.
I'm located in north-east California where lower temps in winter can reach 10F (-12C).  To help resolve the temp issues I have heating mats installed in the battery packs and power to the mats is controlled via a thermostat controlled switch.  Last winter this worked very well and kept the pack temps above 32F (0C).  Although at that time I had only 1 parallel pack (12V system) installed for a short time ... it was pulled out of service to for another project.

It should also be noted that I build all my packs in 12V configuration and installed in group 27 battery boxes.  The 48V systems use 4 of these 12V packs in series.  The BMS (and the active balancer) sample wires are connected to the individual 12V packs via 5 pin JST connectors (1 negative + 4 positive for each cell) that I mount in the battery box lid.  The BMS negative sample wire is connected to only 1 pack.  This allows me flexibility in replacing cells that might go bad by replacing the 12V pack containing the bad cell.

I hope this helps.

**Reply by transistorgit** [reply]
since the latest version, we've implemented a manual "force charge off" and "force discharge off" switch on the dalys. You can also trigger that by node-red and dbus.
So now you should have everything in place to build your own failure handling.

- if one module fails, automatically trigger "force charge/discharge off" by node-red. This prevents from automatic and potentially harmful reconnect.
- fix the problem
- charge modules to same level
- switch off the "force charge/discharge off" function again and the module should connect again.
(be carefull that a power cycled bms of a fixed module may have forgotten the "force off" setting)

**Reply by CaptKrisp** [reply]
Most excellent, thank you!  I shall definitely utilize those functions.

**Reply by SlapJackNpNp** [reply]
@transistorgit  hey, sorry to bother you again. Just wanted to ask what aggregator you are using atm for your system. I saw that you joined an issue for the Dr. Gigavolt aggregator, do you by chance know if dr. gigavolt aggregator works with dalys with the newest dbus driver? Or are you using pulqueros aggregator? Thank you very much

**Reply by transistorgit** [reply]
Hi,

I use the Dr. Gigavolt aggregator all the time. 

Brgds
